Updated version Osprey

P-VS opened the pull request:
Dear,

The last few months I continued using Osprey for the analysis of my spectro data and made some improvements to the code. I herby share the changes I made.

First of all, I solved some problems with the GE nifti files and the Matlab path. Secondly, I made the program a little more user-friendly by:
* Adding the buttons from the start screen (Load Job file, Load MRSCont,…) to the interactive screen
* In the interactive screen, you can upload a JobFile. The fields will be updated with the values specified in the JobFile
* When start Osprey with the Matlab command ‘Osprey(‘start’)’, the program directly opens in the interactive screen rather than in the start screen with the buttons. To avoid too much interference with what you are doing, you can still start Osprey in the normal start screen with the command ‘Osprey’.
* I added a button ’New analysis’ to the main screen. This way you can start a new analysis without the need to first close and restart the program.
* As most users don’t save their data in a program folder (not recommendable) but somewhere on a data disk or server, I changed the default folder for data selection to the working directory set in the main Matlab screen rather than the ‘Osprey/exampledata’ folder. As such, data selection is more easy if the study data is not stored in the exampledata folder by setting the Matlab’s working directory to the datafolder.

Thirdly, I added the analysis of spectro data taken from the Braino phantom to the program. However, today, it will only work for PRESS data taken on a GE scanner, since that is the only data I have. For this, I added a parameter ‘MRS Protocol’. When setting this parameter to ‘Braino phantom’, the program will change the other parameters like ‘Reference frequency’ and ‘Selected Metabolites’ to the default values for phantom data.

At last, I added the GE’s newest pfile version 28.002 to GEload.

I tried to update my version with your changes and to update conflicts, but please check yourself.

Kind regards,
Peter

hezoe100 commented:
Dear Peter,

Thanks for sharing your work on Osprey with us. Your continuous contributions are most appreciated!

I've redirected your commit into its own branch for further testing before merging it into the main develop branch.

Best,
Helge

hezoe100 commented:
Dear Peter (@P-VS) ,

I've been testing your contributions. I do have some questions about your contributions:

- CreateOspreyJob.mlapp and CreateOspreyJob.m seem to be inconsistent or at least I can not start the app on my machine, because of the osprey image. It seems to be defined in the .m file, but not in the .mlapp file. Maybe you need to push the .mlapp file again. I was able to resolve the problem by commenting the image part out for now.

- Could you elaborate a bit more on your braino protocol flag. The reason I'm asking is that we do already have a phantom data scenario flag. This flag changes the referencing step, adds some line broadening , and changes the HSVD filter. What was your reasoning to not opt for the phantom switch? I'd like to streamline the phantom processing and I could integrate your changes into the normal phantom flag. In this case, I'd add braino phantom as a metabolite list option instead of adding the new protocol flag to reduce the number of flags in the MRS container. 

The rest of your changes work like a charm!

Thanks for your contributions,
Helge

P-VS commented:
Dear Helge,

I removed the image and created a new .mlapp and .m file. You find the updated files in my own copy of Osprey: https://github.com/P-VS/osprey/tree/develop/job

The reasoning that I added the braino protocol tag was that I encountered problems when I tried to analyse my phantom data. The program tried to analyse my data as brain data and since your tag wasn't present in the example job.m file I used, I defined a new tag. Later on, when I found your tag appearing in the code, I still got wrong results. Since I am hesitant to make changes in the code that can interfere with what other contributors are doing, I kept using my own flag for this version. Combining both flags and their use in a later stage seems logic of course.
Separating the processing streamline for phantoms from those from brain data as much as possible seemed to be the best approach to solve it. First of all because it is not only a matter of shifting the reference frequency, changing the filter settings and changing the metabolite list. In my code, the flag already changes the reading of the pfile (e.g. the definition of the ppm axis). I had to make new reference metabolite spectra simulations because the relative frequency shifts of the individual metabolites is different from those in brain spectra. Moreover, to my experience from previous studies, likely, also the T1 and T2 values of the metabolites used for the metabolite concentration quantifications differ between phantom and brain signals. But it is still on my to do list to check that with measured data. So, processing phantom data differs from the first till the last step and defining it as a separate protocol feels more natural. Secondly, by defining it as a separate protocol for this particular phantom, it will be more easy to add different protocols for other phantoms with different metabolite compositions (e.g. the MRS phantoms from Philips or Siemens) to the program.

Kind regards,
Peter
